### DISCUSSION TITLE
How to get Hono working with KV?

### DISCUSSION THREAD

**Discussion by g-wozniak** [honojs/hono#1637] [Q&A]
Hello Community,

I am so far amazed by Hono and Cloudflare Workers but I am still new to it.
I am trying to get my KV working. 

I followed this [documentation fragment](https://hono.dev/api/context#env) and [this](https://developers.cloudflare.com/kv/learning/kv-bindings/) but I am probably making some obvious mistake.

**Expected behaviour**

I would expect my KV variables to show once deployed to production but the response is always an empty object. I can see only the local variables correctly (vars) (if I list: `c.env`) but no SECRETS. 

My KV bucket contains the following pair:
`HELLO => value`

The response I am expecting is:
`{"message":{
     "HELLO": "value"
}}`

The response I am getting is:
`{"message":{}}`

_wrangler_
```
name = "somename-api"
compatibility_date = "2023-01-01"

[vars]
ENVIRONMENT = "local"
MY_VARIABLE = "local"

[env.production]
vars = { ENVIRONMENT = "production", MY_VARIABLE="production" }
route = { pattern = "api.stuff.com", custom_domain = true }
kv_namespaces = [
  { binding = "SECRETS", id = "XXXXXXX..." }
]
```

_handler_
```ts
(c: Context<Env>) => {
      const hello = c.env.SECRETS
      return c.json({
         message: hello
      })
}
```

Also, am I right to assume I cannot plug the KV on my local machine in the way below and expect it to work when executing "yarn dev"?
```
[[kv_namespaces]]
binding = "SECRETS"
id = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
```

**Comment by yusukebe** [comment]
This is not Hono matter, so you should not ask this place. But, I think you should set the variables in the Cloudflare Dashboard.

**Reply by g-wozniak** [reply]
The binding is linked fine as Hono states in the debug mode. I can get this work in adequate Workers implementation too.

I realised the mistake in my code in Hono which was caused by assigning wrongly the type to the binding in TypeScript. This caused me not to use await and .get(). 

Spotting the mistake above was probably 5sec with a little empathy considering there is no example in the docs showing how to get value from KV. To me the one purpose of the community is to help when someone struggles; not solely of course but dismissing is always easy. If works for one then he advocates the solution further and it’s how becomes well known. Don’t take me wrong, I am grateful for any docs and I appreciate the work put into it. 

Now I know it’s exactly the same as with Workers but it’s not obvious when one starts. I thought Hono has its own types for bindings.

Still, the values of the KV I am obtaining via Hono are invalid, while using Wrangler are ok. In Hono seem cached or outdated. In the cloudflare dashboard, if I set them manually then in Hono they’re still the old ones showing. I hope I will find the way to fix that too and yes, it’s probable may be nothing to do with Hono.
